ha/util/consul_kv_store.py

```diff
@@ -163,7 +163,7 @@ def get(self, key: str = ""):
             key_val[key['Key']] = key['Value'].decode("utf-8") if isinstance(key['Value'], bytes) else key['Value']
         return key_val
 
-    def delete(self, key: str = ""):
+    def delete(self, key: str = "", recurse=True):
```

Comment:
recurse has always been true at all call points, so why do we need that as parameter ?

Reply:
It is code refactor only, logic and call is still same
In some case may need to delete single key only so there it can be useful

eg:
a/b/c
a/b/c

delete(a/b) will delete all keys
delete(a/b, recurse=False) will not delete keys